### PR TITLE
PDE migration to bullseye

### PR DESCRIPTION
**Pull request by geans-pin**
#### Why I did it
Upgrade docker-pde to bullseye

#### How to verify it
Check Azp status

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: dockers/docker-pde/Dockerfile.j2 [hunk 1]
```diff
@@ -20,8 +20,7 @@ RUN apt-get update && apt-get install -y ipmitool telnet busybox kmod libpcap-de
 RUN apt-get install -y bzip2 curl ioping openssl bc sysstat
 
 # Packages for PDDF
-
-RUN apt-get install -y python-jsonschema
+RUN apt-get install -y python3-jsonschema i2c-tools
```
Comment (qiluo-msft) [inline dockers/docker-pde/Dockerfile.j2:23]: >i2c-tools [](http://example.com/codeflow?start=42&length=9)

Why need this for migration?

File: dockers/docker-pde/docker_init.sh [hunk 2]
```diff
@@ -10,6 +11,8 @@ start_bcm()
 
 PLATFORM_DIR=/usr/share/sonic/platform
 HWSKU_DIR=/usr/share/sonic/hwsku
+# Default use python2 version
+SONIC_PLATFORM_API_PYTHON_VERSION=2
```
Comment (saiarcot895) [inline dockers/docker-pde/docker_init.sh:15]: Why is this defaulting to Python 2? Also, where is `SONIC_PLATFORM_API_PYTHON_VERSION` used?

File: dockers/docker-pde/Dockerfile.j2 (hunk 1 above)
Comment (geans-pin) [reply dockers/docker-pde/Dockerfile.j2:23]: This is required for PDDF support

File: dockers/docker-pde/docker_init.sh (hunk 2 above)
Comment (geans-pin) [reply dockers/docker-pde/docker_init.sh:15]: This is just default value. If you see the following lines, the value will be assigned to new version.

Geans


# If the Python 3 sonic-platform package is not installed, try to install it
python3 -c "import sonic_platform" > /dev/null 2>&1 || pip3 show sonic-platform > /dev/null 2>&1
if [ $? -ne 0 ]; then
    SONIC_PLATFORM_WHEEL="/usr/share/sonic/platform/sonic_platform-1.0-py3-none-any.whl"
    echo "sonic-platform package not installed, attempting to install..."
    if [ -e ${SONIC_PLATFORM_WHEEL} ]; then
       pip3 install ${SONIC_PLATFORM_WHEEL}
       if [ $? -eq 0 ]; then
          echo "Successfully installed ${SONIC_PLATFORM_WHEEL}"
          SONIC_PLATFORM_API_PYTHON_VERSION=3
       else
          echo "Error: Failed to install ${SONIC_PLATFORM_WHEEL}"
       fi
    else
       echo "Error: Unable to locate ${SONIC_PLATFORM_WHEEL}"
    fi
else
    SONIC_PLATFORM_API_PYTHON_VERSION=3
fi

**Comment by adyeung** [comment]
@qiluo-msft @saiarcot895 pls help signoff and merge if there is no further comments

File: dockers/docker-pde/docker_init.sh (hunk 2 above)
Comment (saiarcot895) [reply dockers/docker-pde/docker_init.sh:15]: All successful code paths are setting `SONIC_PLATFORM_API_PYTHON_VERSION` to 3 anyways, so I'd recommend just setting `SONIC_PLATFORM_API_PYTHON_VERSION=3` here.

**Review by qiluo-msft — APPROVED** [review]
LGTM. Please check with other reviewers.

File: dockers/docker-pde/docker_init.sh (hunk 2 above)
Comment (geans-pin) [reply dockers/docker-pde/docker_init.sh]: Done,  change default to 3 now

**Review by saiarcot895 — APPROVED** [review]

**Pull request merged**